### ISSUE TITLE
DatePicker selects wrong month

### ISSUE THREAD

**Issue by pmiklashevich** [bryntum/support#1511]
[Forum post](https://www.bryntum.com/forum/viewtopic.php?f=52&t=15261&p=76736)

**IMPORTANT**: this ticket is blocked by #1563

http://lh/bryntum-suite/grid/examples/columntypes/
1. Edit a date cell
2. Select the 31st of December and finish editing.
3. Edit the cell again.
4. Select June in the date picker
5. See June is changed with July. (BUG)
Sometimes it's reproducible from the second attempt. So need to edit the cell again.
![Запись активности на экране 2020-09-14 в 11 11 01](https://user-images.githubusercontent.com/57486733/93060823-5a066400-f67b-11ea-9d03-a6714a7f980b.gif)

**Comment by pmiklashevich** [comment]
DatePicker.onMonthPicked should be:
```
    onMonthPicked({ value, oldValue }) {
        this.activeDate = DateHelper.add(this.activeDate, value - oldValue, 'month');
    }
```

**Comment by pmiklashevich** [comment]
This is the test. But the problem is that "onMonthPicked" is called twice. Going to create a separate ticket for that. When the ticket is done and this [PR](https://github.com/bryntum/bryntum-suite/pull/2035) created by @ExtAnimal is merged to master, feel free to add this test to the DatePicker.t
```
    // https://github.com/bryntum/support/issues/1511
    t.it('Should select ', t => {
        datePicker = new DatePicker({
            date        : new Date(2020, 11, 31),
            value       : new Date(2020, 11, 31),
            activeDate  : new Date(2020, 11, 31),
            editOnHover : false,
            appendTo    : document.body,
            autoShow    : true
        });

        const spy = t.spyOn(datePicker, 'onMonthPicked');

        t.chain(
            { click : '.b-datepicker-month' },
            { click : '.b-list-item:contains(June)' },
            () => {
                // June has 30 days
                t.is(datePicker.activeDate, new Date(2020, 5, 30));
                t.expect(spy).toHaveBeenCalled(1);
            }
        );
    });
```
